src/CheckStrings.jl

```diff
@@ -0,0 +1,200 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+## Functions to check validity of UTF-8, UTF-16, and UTF-32 encoded strings,
+#  and also to return information necessary to convert to other encodings
+
+module CheckStrings
+
+using Encodings
+
+export check_string
```

Comment:
I worry though that `check` might be too general a name... this is really just to check string encodings, stored in various ways...